exdir/core/mode.py

```diff
@@ -0,0 +1,32 @@
+from enum import Enum
+
+
+class OpenMode(Enum):
+    READ_WRITE = 1
+    READ_ONLY = 2
+    FILE_CLOSED = 3
+
+
+def assert_file_open(func):
```

Comment:
These are nice, but I do not think we gain anything by having them as decorators. You could just call this function at the top of each of the now-decorated functions (just remove the wrapper and simply raise if there is an issue). That is more readable to the average Python developer and easier to look up by following function definitions in IDEs.